rslib/src/text.rs

```diff
@@ -380,22 +380,61 @@ pub(crate) fn ensure_string_in_nfc(s: &mut String) {
     }
 }
 
+static EXTRA_NO_COMBINING_REPLACEMENTS: phf::Map<char, &str> = phf::phf_map! {
+'€'  =>  "E",
+'Æ'  =>  "AE",
+'Ð'  =>  "D",
+'Ø'  =>  "O",
+'Þ'  =>  "TH",
+'ß'  =>  "s",
+'æ'  =>  "ae",
+'ð'  =>  "d",
+'ø'  =>  "o",
+'þ'  =>  "th",
+'Đ'  =>  "D",
+'đ'  =>  "d",
+'Ħ'  =>  "H",
+'ħ'  =>  "h",
+'ı'  =>  "i",
+'ĸ'  =>  "k",
+'Ł'  =>  "L",
+'ł'  =>  "l",
+'Ŋ'  =>  "N",
+'ŋ'  =>  "n",
+'Œ'  =>  "OE",
+'œ'  =>  "oe",
+'Ŧ'  =>  "T",
+'ŧ'  =>  "t",
+'Ə'  =>  "E",
+'ǝ'  =>  "e",
+'ɑ'  =>  "a",
+};
+
 /// Convert provided string to NFKD form and strip combining characters.
 pub(crate) fn without_combining(s: &str) -> Cow<str> {
     // if the string is already normalized
     if matches!(is_nfkd_quick(s.chars()), IsNormalized::Yes) {
         // and no combining characters found, return unchanged
-        if !s.chars().any(is_combining_mark) {
+        if !s
+            .chars()
+            .any(|c| is_combining_mark(c) || EXTRA_NO_COMBINING_REPLACEMENTS.contains_key(&c))
+        {
             return s.into();
         }
     }
 
     // we need to create a new string without the combining marks
-    s.chars()
+    let mut s = s
+        .chars()
         .nfkd()
         .filter(|c| !is_combining_mark(*c))
-        .collect::<String>()
-        .into()
+        .collect::<String>();
```

Comment:
Use of phf is nice! I think the step below could be done more efficiently though, by using .map() here to transform the unwanted characters, instead of modifying the string below. You should be able to look up each character instead of having to iterate over the phf, too.

Reply:
(might be more efficient to do this in a 'for char in ... { output.push(char)  or output.push_str(repl) }' than using flat_map)